### PR TITLE
Setup Database and schemas

### PR DESCRIPTION
**Pull request by westonmiller**
#### What does it do?

Sets mongoldb database using Mongoose
Adds RESTful APIs for:
`/beers`  post, get
`/beers/{id}` get, delete
`/beers/{id}/reviews` post, get
`/beers/{id}/reviews` get
#### Any helpful background information?

Need to setup mongoose on local machine for testing.
#### New dependencies? What are they used for?

mongoose: Object modeling for Nodejs and MongoDB
bodyParser: allows easy parsing of html requests.
#### Definition of Done
- [x] has no tests

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,8 +1,138 @@
...
+
+app.post('/beers', (request, response) => {
+  const {name, brewery, submitter, description} = request.body;
+
+  Beer.find(
+    {
+      name: name.toLowerCase(),
+      brewery: brewery.toLowerCase()
+    }, (error, beers) => {
+    if (beers.length) {
```
Comment (reedsa) [inline server.js]: I think this block should be indented one more.

File: server.js [hunk 2]
```diff
@@ -1,8 +1,138 @@
...
+      response.status(404).send('No Beer Found');
+    }
+  });
+});
+
+app.delete('/beers/:id', (request, response) => {
+  const { id } = request.params;
+
+  Beer.findById(id, (error, beer) => {
+    beer.remove((error) => {
```
Comment (reedsa) [inline server.js]: I'm not sure if this deletes the reviews for the beer being removed, does it?

File: server.js [hunk 3]
```diff
@@ -1,8 +1,138 @@
...
+        } else {
+          response.send(error);
+        }
+      });
+    }
+  });
+});
+
+app.get('/beers', (request, response) => {
+  Beer.find({}).populate('reviews').exec((error, beers) => {
```
Comment (reedsa) [inline server.js:45]: So I'm not clear on what `populate` does. Is it returning the beers and all the reviews for each beer?

**Review by reedsa — CHANGES_REQUESTED** [review]
Looks like fun. I'm not sure what you decided to do about documentation but the API calls aren't too complicated. Just made a couple comments, but overall it seems good to ship!

File: server.js (hunk 3 above)
Comment (westonmiller) [reply server.js:45]: The way it stores the relationship is just by id, using populate, will actually return the review objects, not just their ids

File: server.js (hunk 2 above)
Comment (westonmiller) [reply server.js]: it does not, it leaves the reviews, I think

**Pull request merged**